### DISCUSSION TITLE
Are there any operations for which airflow integration extracts output/input ?

### DISCUSSION THREAD

**Discussion by cerveada** [OpenLineage/OpenLineage#696] [Q&A]
I am trying to produce open lineage message with an output/input using airflow integration. I am using **Airflow 2.2.4**.

I noticed there is #675. But is there any SQL command (or other Operator) for which the extraction works right now?

I tried several things using `PostgresOperator`, but output/input is always empty.

**Comment by mobuchowski** [comment]
Do you see any particular errors/exceptions? If not, can you provide SQL used in the query?

>But is there any SQL command (or other Operator) for which the extraction works right now?

BigQuery extractor gets input and output tables from BigQuery backend, it does not rely on SQL parser to get datasets.

**Reply by cerveada** [reply]
There are no exceptions, it simply does not extract the output/input

For example, this task:
```python
import datetime
import pendulum

import requests
from airflow.decorators import dag, task
from airflow.providers.postgres.hooks.postgres import PostgresHook
from airflow.providers.postgres.operators.postgres import PostgresOperator

@dag(
    schedule_interval="0 0 * * *",
    start_date=pendulum.datetime(2021, 1, 1, tz="UTC"),
    catchup=False,
    dagrun_timeout=datetime.timedelta(minutes=60),
)
def InsertDataDag():
    PostgresOperator(
        task_id="new_room_booking",
        postgres_conn_id="local-postgre",
        sql="""
            DELETE FROM public."Employees";
            INSERT INTO public."Employees" VALUES (1, 'TALES OF SHIVA', 'Mark', 'mark', 0);
            """
    )    


dag = InsertDataDag()
```

**Reply by cerveada** [reply]
> BigQuery extractor gets input and output tables from BigQuery backend, it does not rely on SQL parser to get datasets.

I was hoping for something that could be run locally on my machine, but thanks for info.

**Reply by collado-mike** [reply]
I think the issue is that there are multiple SQL statements in your operator. Try splitting the statements into two operators and I think you'll see everything work.

**Reply by cerveada** [reply]
I did few more tries and this works:
```
"""INSERT INTO public.foo VALUES ('bar')"""
```
but this doesn't:
```
"""INSERT INTO public."foo" VALUES ('bar')"""
```
So it seems the output extraction fails when quotes are used.

**Comment by mobuchowski** [comment]
As far as I can see, all those problems will be solved by https://github.com/OpenLineage/OpenLineage/pull/644
